CorsixTH/Lua/hospital.lua

```diff
@@ -1136,8 +1136,13 @@ function Hospital:resolveEmergency()
   local rescued_patients = emer.cured_emergency_patients
   for _, patient in ipairs(self.emergency_patients) do
     if patient and not patient.cured and not patient.dead
-        and not patient.going_home and not patient:getRoom() then
-      patient:die()
+        and not patient.going_home then
+      if not patient:getRoom() then
```

Comment:
patient.lua has an extra condition here:

```lua
 if not self:getRoom() and not self:getCurrentAction().is_leaving then
      self:setMood("dying5", "deactivate")
      self:die()
    end
```

I also wonder why this isn't considered at all for Patient:treatDisease which calls die() inside a room without a check.

Patient:die itself appears to handle being inside a room.

```lua
  if self:getRoom() then
    self:queueAction(MeanderAction():setCount(1))
  else
    self:setNextAction(MeanderAction():setCount(1))
  end
```

What I'm saying is this condition seems suspect to me; I'm not sure the motivation for having room checked in some places and not others. My best guess is to give the treatment a chance of being successful.

Reply:
@ARGAMX if you're also working on this consider the same question. It might be all right as is.